integration/command.go

```diff
@@ -0,0 +1,251 @@
+	command *exec.Cmd
+
+	// stdout contains command standard output when started using Startcommand().
+	stdout bytes.Buffer
+
+	// stderr contains command standard output when started using Startcommand().
+	stderr bytes.Buffer
+
+	// expectedString contains the exact expected output of the command.
+	expectedString string
+
+	// expectedRegexp contains a regex used to match against the command output.
+	expectedRegexp string
+
+	// cleanup indicates this command is used to clean resource and should not be
+	// skipped even if previous commands failed.
+	cleanup bool
+
+	// startAndStop indicates this command should first be started then stopped.
+	// It corresponds to gadget like execsnoop which wait user to type Ctrl^C.
+	startAndStop bool
+
+	// started indicates this command was started.
+	// It is only used by command which have startAndStop set.
+	started bool
+}
+
+var deployInspektorGadget *command = &command{
+	name:           "Deploy Inspektor Gadget",
+	cmd:            "$KUBECTL_GADGET deploy $GADGET_IMAGE_FLAG | kubectl apply -f -",
+	expectedRegexp: "gadget created",
+}
+
+var waitUntilInspektorGadgetPodsDeployed *command = &command{
+	name: "Wait until the gadget pods are started",
+	cmd:  "for POD in $(sleep 5 ; kubectl get pod -n kube-system -l k8s-app=gadget -o name) ; do kubectl wait --timeout=30s -n kube-system --for=condition=ready $POD ; done ; kubectl get pod -n kube-system",
+}
+
+var waitUntilInspektorGadgetPodsInitialized *command = &command{
+	name: "Wait until Inspektor Gadget is initialised",
+	cmd:  "sleep 15",
+}
+
+var createTestNamespace *command = &command{
+	name:           "Create test namespace",
+	cmd:            "kubectl create ns test-ns",
+	expectedString: "namespace/test-ns created\n",
+}
+
+var waitUntilTestPodReady *command = &command{
+	name:           "Wait until test pod ready",
+	cmd:            "kubectl wait pod --for condition=ready -n test-ns test-pod",
+	expectedString: "pod/test-pod condition met\n",
+}
+
+var deleteTestPod *command = &command{
+	name:           "Delete test pod",
+	cmd:            "kubectl delete pod -n test-ns test-pod",
+	expectedString: "pod \"test-pod\" deleted\n",
+	cleanup:        true,
+}
+
+var cleanupTestNamespace *command = &command{
+	name:           "cleanup test namespace",
+	cmd:            "kubectl delete ns test-ns",
+	expectedString: "namespace \"test-ns\" deleted\n",
+	cleanup:        true,
+}
+
+var cleanupInspektorGadget *command = &command{
+	name:           "cleanup gadget deployment",
+	cmd:            "$KUBECTL_GADGET deploy $GADGET_IMAGE_FLAG | kubectl delete --wait=false -f -",
+	expectedRegexp: "\"gadget\" deleted",
+	cleanup:        true,
+}
+
+// run runs the command on the given as parameter test.
+// It returns true to indicate the command failed, false if it succeeded.
+func (c *command) run(t *testing.T, failed bool) bool {
+	t.Run(c.name, func(t *testing.T) {
+		if failed && !c.cleanup {
+			t.Skip("Previous c failed.")
+		}
```

Comment:
If we remove the t.Run call, then we can get rid of the `failed` variable everywhere, it's not needed.